### PR TITLE
Adds visualisation of yaml state machine definitions

### PR DESCRIPTION
**Pull request by alanbo**
## Description
Basic visualization capability for state machines defined in yaml.  

## Motivation and Context
We want to give users basic capability to visualize valid state machines in yaml. This won't be an official release and we will not provide errors or linting when the state machine definition is incorrect. 

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts [hunk 1]
```diff
@@ -99,7 +108,27 @@ export class AslVisualization {
         )
 
         const sendUpdateMessage = async (updatedTextDocument: vscode.TextDocument) => {
-            const isValid = await isDocumentValid(updatedTextDocument)
+            const isYAML = updatedTextDocument.languageId === 'yaml'
+            const text = updatedTextDocument.getText()
+            let stateMachineData = text
```
Comment (awschristou) [inline src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts:113]: does this stay empty when isYAML is false? Looks like that might fail isDocumentValid check

File: src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts [hunk 2]
```diff
@@ -99,7 +108,27 @@ export class AslVisualization {
         )
 
         const sendUpdateMessage = async (updatedTextDocument: vscode.TextDocument) => {
-            const isValid = await isDocumentValid(updatedTextDocument)
+            const isYAML = updatedTextDocument.languageId === 'yaml'
```
Comment (awschristou) [inline src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts:111]: nit: isYaml

File: package.json [hunk 3]
```diff
@@ -934,7 +934,8 @@
         "vue": "^2.5.16",
         "winston": "^3.2.1",
         "winston-transport": "^4.3.0",
-        "xml2js": "^0.4.19"
+        "xml2js": "^0.4.19",
+        "yaml": "^1.9.2"
```
Comment (awschristou) [inline package.json:938]: There is already a package reference to `js-yaml`, is it necessary to add another yaml package reference?

File: src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts [hunk 4]
```diff
@@ -99,7 +108,27 @@ export class AslVisualization {
...
+            const text = updatedTextDocument.getText()
+            let stateMachineData = text
+            let yamlErrors: string[] = []
+
+            if (isYAML) {
```
Comment (awschristou) [inline src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts:116]: Please ensure there are tests for both yaml and json inputs to this function

File: package.json (hunk 3 above)
Comment (alanbo) [reply package.json:938]: From what I see, the existing package will allow user to use merges, aliases and other things (not sure exactly what) specified in 1.2 without throwing any errors. The user will think it is fine until they actually try to deploy it. At least according to the documentation, cloud formation / sam will error on encountering it.

Comment (awschristou) [reply package.json:938]: Thank you for pointing this out. I've made a work item to change the existing parsing code away from that 1.2 spec later on.

Comment (awschristou) [reply package.json:938]: It seems reasonable to add the yaml package, and we will remove the other package when we can

File: src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts (hunk 1 above)
Comment (alanbo) [reply src/stepFunctions/commands/visualizeStateMachine/aslVisualization.ts:113]: State machine data is a text of `updatedTextDocumen` - there is an assignment there. If it is a yaml file this variable get reassigned with a new value.

**Comment by kiiadi** [comment]
Might be good to get some screen-shots on the PR.

**Comment by alanbo** [comment]
> Might be good to get some screen-shots on the PR.

This update doesn't introduce any visual changes. The only difference is that user will be visualize yaml state machine from command palette.

**Pull request merged**